khmer/utils.py

```diff
@@ -18,9 +18,89 @@ def print_error(msg):
     print >>sys.stderr, msg
 
 
+def check_is_pair(record1, record2):
+    """
+    Checks to see if the two sequence records are left and right pairs
+    of each other, respectively.  Returns True or False as appropriate.
+
+    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
+    and 'seq::... 2::...'.
+    """
+    is_fastq = False
+    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
+        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
+            raise ValueError("both records must be same type (FASTA or FASTQ)")
+        is_fastq = True
```

Comment:
On Tue, Feb 10, 2015 at 02:10:33PM -0800, Camille Scott wrote:
> > @@ -18,9 +18,89 @@ def print_error(msg):
> >      print >>sys.stderr, msg
> >  
> >  
> > +def check_is_pair(record1, record2):
> > +    """
> > +    Checks to see if the two sequence records are left and right pairs
> > +    of each other, respectively.  Returns True or False as appropriate.
> > +
> > +    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
> > +    and 'seq::... 2::...'.
> > +    """
> > +    is_fastq = False
> > +    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
> > +        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
> > +            raise ValueError("both records must be same type (FASTA or FASTQ)")
> > +        is_fastq = True
> 
> this is erroneous for `khmer.ReadParser`, which *always* has an accuracy attribute, even for FASTA (it just leaves the field as an empty string)

Yes, I know.  We should fix that one way or the other (personally I prefer
the screed approach, but only at a +0 level).  Neither decision was made
with any forethought.